src/WP_CLI/CommandWithUpgrade.php

```diff
@@ -771,4 +803,88 @@ private function parse_url_host_component( $url, $component ) {
 		// phpcs:ignore WordPress.WP.AlternativeFunctions.parse_url_parse_url -- parse_url will only be used in absence of wp_parse_url.
 		return function_exists( 'wp_parse_url' ) ? wp_parse_url( $url, $component ) : parse_url( $url, $component );
 	}
+
+	/**
+	 * Get the latest package version based on a given repo slug.
+	 *
+	 * @param string $repo_slug
+	 *
+	 * @return array{ name: string, url: string }|\WP_Error
+	 */
+	public function get_the_latest_github_version( $repo_slug ) {
+		$api_url = sprintf( $this->github_releases_api_endpoint, $repo_slug );
+		$token   = getenv( 'GITHUB_TOKEN' );
+
+		$request_arguments = $token ? [ 'headers' => 'Authorization: Bearer ' . getenv( 'GITHUB_TOKEN' ) ] : [];
+
+		$response = \wp_remote_get( $api_url, $request_arguments );
+
+		if ( \is_wp_error( $response ) ) {
+			return $response;
+		}
+
+		$body         = \wp_remote_retrieve_body( $response );
+		$decoded_body = json_decode( $body );
+
+		if ( wp_remote_retrieve_response_code( $response ) === \WP_Http::FORBIDDEN ) {
+			return new \WP_Error(
+				\WP_Http::FORBIDDEN,
```

Comment:
Looks like `WP_Http::FORBIDDEN` doesn't exist yet on WordPress 3.7, see https://github.com/wp-cli/extension-command/actions/runs/8851555744/job/24308388075?pr=421#step:11:67

I suppose we can just hardcode 403 here, maybe adding an explanatory comment

Reply:
Fixed up in fc16a6a5